compiler/src/iree/compiler/GlobalOptimization/ExpandVectors.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===--------------- ExpandVectors.cpp -----------------------------------===//
+// Expands vectors in matrix/vector operations (vecmat, matvec, batch_matvec)
+// into matrices in order to enable tiling.
+//===---------------------------------------------------------------------===//
+
+#include "iree/compiler/GlobalOptimization/PassDetail.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace GlobalOptimization {
+
+namespace {
+
+struct ExpandVectors
+    : public OpInterfaceRewritePattern<linalg::ContractionOpInterface> {
+  using OpInterfaceRewritePattern<
+      linalg::ContractionOpInterface>::OpInterfaceRewritePattern;
+  LogicalResult matchAndRewrite(linalg::ContractionOpInterface op,
+                                PatternRewriter &rewriter) const override {
+    auto linalgOp = dyn_cast<linalg::LinalgOp>(op.getOperation());
+    if (!linalgOp || !linalgOp.hasTensorSemantics())
+      return failure();
```

Comment:
We can simplify the check. IIRC, a ContractionOpInterface must be a `linalg::LinalgOp`.

```suggestion
    if (!linalgOp.hasTensorSemantics())
      return failure();
```

style nit: add braces around the trivial if statement.